File: src/Domain/Models/Starship.cs
namespace Domain.Models;

public class Starship
{
    public string Uid { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public string Manufacturer { get; set; } = string.Empty;
    public string CostInCredits { get; set; } = string.Empty;
    public string Length { get; set; } = string.Empty;
    public string MaxAtmospheringSpeed { get; set; } = string.Empty;
    public string Crew { get; set; } = string.Empty;
    public string Passengers { get; set; } = string.Empty;
    public string CargoCapacity { get; set; } = string.Empty;
    public string Consumables { get; set; } = string.Empty;
    public string HyperdriveRating { get; set; } = string.Empty;
    public string MGLT { get; set; } = string.Empty;
    public string StarshipClass { get; set; } = string.Empty;
    public List<string> Pilots { get; set; } = new();
    public List<string> Films { get; set; } = new();
    public DateTime Created { get; set; }
    public DateTime Edited { get; set; }
    public string Url { get; set; } = string.Empty;
}

File: src/Frontend/swapi-frontend/src/app/models/starship.ts
export interface StarshipSummary {
  uid: string;
  name: string;
  url: string;
}

export interface Starship {
  uid: string;
  name: string;
  model: string;
  manufacturer: string;
  costInCredits: string;
  length: string;
  maxAtmospheringSpeed: string;
  crew: string;
  passengers: string;
  cargoCapacity: string;
  consumables: string;
  hyperdriveRating: string;
  mglt: string;
  starshipClass: string;
  pilots: string[];
  films: string[];
  created: Date;
  edited: Date;
  url: string;
}